node-api/handlers/utils/id.go

```diff
@@ -112,14 +112,10 @@ func IsTimestampIDPrefix(timestampID string) bool {
 }
 
 // slotFromStateID returns a slot number from the given state ID.
-// In case of genesis, we will return 1 as the slot number. This is to ensure
-// validator API endpoints return a valid slot number for genesis.
 func slotFromStateID(id string) (math.Slot, error) {
 	switch id {
 	case StateIDFinalized, StateIDJustified, StateIDHead:
 		return Head, nil
-	case StateIDGenesis:
-		return 1, nil
```

Comment:
This is the core issue:
- BeaconKit does not commit genesis state at height zero. Instead it keeps it in memory and commit it along with block 1 changes, at height 1.
- This explains the mapping above but it is not correct as any validator added with block 1 but not in genesis would appear to be in genesis.